mongodb/trigger/eventlistener/trigger.go

```diff
@@ -0,0 +1,313 @@
+package eventlistener
+
+import (
+	"context"
+	"encoding/json"
+
+	//	"github.com/TIBCOSoftware/flogo-lib/core/data"
+	//	"github.com/TIBCOSoftware/flogo-lib/core/trigger"
+	//	"github.com/TIBCOSoftware/flogo-lib/logger"
+
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+	"github.com/project-flogo/core/trigger"
+	mongodb "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+	"go.mongodb.org/mongo-driver/bson"
+	mongo "go.mongodb.org/mongo-driver/mongo"
+)
+
+var triggerMd = trigger.NewMetadata(&Settings{}, &HandlerSettings{}, &Output{})
+
+func init() {
+	_ = trigger.Register(&MyTrigger{}, &MyTriggerFactory{})
+}
+
+// MyTriggerFactory My Trigger factory
+type MyTriggerFactory struct {
+	metadata *trigger.Metadata
+}
+
+//NewFactory create a new Trigger factory
+func NewFactory(md *trigger.Metadata) trigger.Factory {
+	//	fmt.Println("============NewFactory==")
+	return &MyTriggerFactory{metadata: md}
+}
+
+//New Creates a new trigger instance for a given id
+func (t *MyTriggerFactory) New(config *trigger.Config) (trigger.Trigger, error) {
+	s := &Settings{}
+	var err error
+	s.Connection, err = mongodb.GetSharedConfiguration(config.Settings["mongodbConnection"])
+	mConn := s.Connection.(*mongodb.MongodbSharedConfigManager)
+	mclient := mConn.GetClient()
+	// opts := options.Client()
+	// client, err := mongo.NewClient(opts.ApplyURI(mConn.GetClientConfiguration().ConnectionURI))
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	// defer cancel()
+	// err = client.Connect(ctx)
+	// if err != nil {
+	// 	fmt.Println("===connect error==", err)
+	// }
+	// err = client.Ping(ctx, nil)
+	// if err != nil {
+	// 	fmt.Println("===ping error==", err)
+	// } else {
+	// 	fmt.Println("Ping success")
+	// }
+	if err != nil {
+		return nil, err
+	}
+
+	return &MyTrigger{metadata: t.metadata, settings: s, id: config.Id, mclient: mclient}, nil
+	//	return &MyTrigger{metadata: t.metadata, settings: s, id: config.Id}, nil
+}
+
+// Metadata implements trigger.Factory.Metadata
+func (*MyTriggerFactory) Metadata() *trigger.Metadata {
+	return triggerMd
+}
+
+// MyTrigger is a stub for your Trigger implementation
+type MyTrigger struct {
```

Comment:
It's better if it's just `Trigger` rather than `MyTrigger`.

Reply:
Changed the following in trigger.go..
- MyTrigger to Trigger
- MyTriggerFactory to TriggerFactory
- Cleaned up commented out lines